// backendless/es/files/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _applyDecoratedDescriptor2 = _interopRequireDefault(require("@babel/runtime/helpers/applyDecoratedDescriptor"));

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _utils = _interopRequireDefault(require("../utils"));

var _decorators = require("../decorators");

var _persmissions = _interopRequireDefault(require("./persmissions"));

var _save = require("./save");

var _upload = require("./upload");

var _listinig = require("./listinig");

var _rename = require("./rename");

var _move = require("./move");

var _remove = require("./remove");

var _copy = require("./copy");

var _exists = require("./exists");

var _removeDirectory = require("./remove-directory");

var _count = require("./count");

var _dec, _dec2, _dec3, _dec4, _dec5, _dec6, _dec7, _dec8, _dec9, _dec10, _obj, _init, _init2, _init3, _init4, _init5, _init6, _init7, _init8, _init9, _init10;

var Files = function Files(app) {
  (0, _classCallCheck2["default"])(this, Files);
  this.app = app;
  this.Permissions = new _persmissions["default"](app);
};

(0, _extends2["default"])(Files.prototype, (_dec = (0, _decorators.deprecated)('Backendless.Files', 'Backendless.Files.saveFile'), _dec2 = (0, _decorators.deprecated)('Backendless.Files', 'Backendless.Files.upload'), _dec3 = (0, _decorators.deprecated)('Backendless.Files', 'Backendless.Files.listing'), _dec4 = (0, _decorators.deprecated)('Backendless.Files', 'Backendless.Files.renameFile'), _dec5 = (0, _decorators.deprecated)('Backendless.Files', 'Backendless.Files.moveFile'), _dec6 = (0, _decorators.deprecated)('Backendless.Files', 'Backendless.Files.copyFile'), _dec7 = (0, _decorators.deprecated)('Backendless.Files', 'Backendless.Files.remove'), _dec8 = (0, _decorators.deprecated)('Backendless.Files', 'Backendless.Files.exists'), _dec9 = (0, _decorators.deprecated)('Backendless.Files', 'Backendless.Files.removeDirectory'), _dec10 = (0, _decorators.deprecated)('Backendless.Files', 'Backendless.Files.getFileCount'), (_obj = {
  saveFileSync: _utils["default"]["synchronized"](_save.saveFile),
  saveFile: _utils["default"].promisified(_save.saveFile),
  uploadSync: _utils["default"]["synchronized"](_upload.upload),
  upload: _utils["default"].promisified(_upload.upload),
  listingSync: _utils["default"]["synchronized"](_listinig.listing),
  listing: _utils["default"].promisified(_listinig.listing),
  renameFileSync: _utils["default"]["synchronized"](_rename.renameFile),
  renameFile: _utils["default"].promisified(_rename.renameFile),
  moveFileSync: _utils["default"]["synchronized"](_move.moveFile),
  moveFile: _utils["default"].promisified(_move.moveFile),
  copyFileSync: _utils["default"]["synchronized"](_copy.copyFile),
  copyFile: _utils["default"].promisified(_copy.copyFile),
  removeSync: _utils["default"]["synchronized"](_remove.remove),
  remove: _utils["default"].promisified(_remove.remove),
  existsSync: _utils["default"]["synchronized"](_exists.exists),
  exists: _utils["default"].promisified(_exists.exists),
  removeDirectorySync: _utils["default"]["synchronized"](_removeDirectory.removeDirectory),
  removeDirectory: _utils["default"].promisified(_removeDirectory.removeDirectory),
  getFileCountSync: _utils["default"]["synchronized"](_count.getFileCount),
  getFileCount: _utils["default"].promisified(_count.getFileCount)
}, ((0, _applyDecoratedDescriptor2["default"])(_obj, "saveFileSync", [_dec], (_init = Object.getOwnPropertyDescriptor(_obj, "saveFileSync"), _init = _init ? _init.value : undefined, {
  enumerable: true,
  configurable: true,
  writable: true,
  initializer: function initializer() {
    return _init;
  }
}), _obj), (0, _applyDecoratedDescriptor2["default"])(_obj, "uploadSync", [_dec2], (_init2 = Object.getOwnPropertyDescriptor(_obj, "uploadSync"), _init2 = _init2 ? _init2.value : undefined, {
  enumerable: true,
  configurable: true,
  writable: true,
  initializer: function initializer() {
    return _init2;
  }
}), _obj), (0, _applyDecoratedDescriptor2["default"])(_obj, "listingSync", [_dec3], (_init3 = Object.getOwnPropertyDescriptor(_obj, "listingSync"), _init3 = _init3 ? _init3.value : undefined, {
  enumerable: true,
  configurable: true,
  writable: true,
  initializer: function initializer() {
    return _init3;
  }
}), _obj), (0, _applyDecoratedDescriptor2["default"])(_obj, "renameFileSync", [_dec4], (_init4 = Object.getOwnPropertyDescriptor(_obj, "renameFileSync"), _init4 = _init4 ? _init4.value : undefined, {
  enumerable: true,
  configurable: true,
  writable: true,
  initializer: function initializer() {
    return _init4;
  }
}), _obj), (0, _applyDecoratedDescriptor2["default"])(_obj, "moveFileSync", [_dec5], (_init5 = Object.getOwnPropertyDescriptor(_obj, "moveFileSync"), _init5 = _init5 ? _init5.value : undefined, {
  enumerable: true,
  configurable: true,
  writable: true,
  initializer: function initializer() {
    return _init5;
  }
}), _obj), (0, _applyDecoratedDescriptor2["default"])(_obj, "copyFileSync", [_dec6], (_init6 = Object.getOwnPropertyDescriptor(_obj, "copyFileSync"), _init6 = _init6 ? _init6.value : undefined, {
  enumerable: true,
  configurable: true,
  writable: true,
  initializer: function initializer() {
    return _init6;
  }
}), _obj), (0, _applyDecoratedDescriptor2["default"])(_obj, "removeSync", [_dec7], (_init7 = Object.getOwnPropertyDescriptor(_obj, "removeSync"), _init7 = _init7 ? _init7.value : undefined, {
  enumerable: true,
  configurable: true,
  writable: true,
  initializer: function initializer() {
    return _init7;
  }
}), _obj), (0, _applyDecoratedDescriptor2["default"])(_obj, "existsSync", [_dec8], (_init8 = Object.getOwnPropertyDescriptor(_obj, "existsSync"), _init8 = _init8 ? _init8.value : undefined, {
  enumerable: true,
  configurable: true,
  writable: true,
  initializer: function initializer() {
    return _init8;
  }
}), _obj), (0, _applyDecoratedDescriptor2["default"])(_obj, "removeDirectorySync", [_dec9], (_init9 = Object.getOwnPropertyDescriptor(_obj, "removeDirectorySync"), _init9 = _init9 ? _init9.value : undefined, {
  enumerable: true,
  configurable: true,
  writable: true,
  initializer: function initializer() {
    return _init9;
  }
}), _obj), (0, _applyDecoratedDescriptor2["default"])(_obj, "getFileCountSync", [_dec10], (_init10 = Object.getOwnPropertyDescriptor(_obj, "getFileCountSync"), _init10 = _init10 ? _init10.value : undefined, {
  enumerable: true,
  configurable: true,
  writable: true,
  initializer: function initializer() {
    return _init10;
  }
}), _obj)), _obj)));
var _default = Files;
exports["default"] = _default;
