// backendless/lib/commerce/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _applyDecoratedDescriptor2 = _interopRequireDefault(require("@babel/runtime/helpers/applyDecoratedDescriptor"));

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _utils = _interopRequireDefault(require("../utils"));

var _decorators = require("../decorators");

var _validatePlayPurchase = require("./validate-play-purchase");

var _cancelPlaySubscription = require("./cancel-play-subscription");

var _getPlaySubscriptionStatus = require("./get-play-subscription-status");

var _dec, _dec2, _dec3, _obj, _init, _init2, _init3;

var Commerce = function Commerce(app) {
  (0, _classCallCheck2["default"])(this, Commerce);
  this.app = app;
};

(0, _extends2["default"])(Commerce.prototype, (_dec = (0, _decorators.deprecated)('Backendless.Commerce', 'Backendless.Commerce.validatePlayPurchase'), _dec2 = (0, _decorators.deprecated)('Backendless.Commerce', 'Backendless.Commerce.cancelPlaySubscription'), _dec3 = (0, _decorators.deprecated)('Backendless.Commerce', 'Backendless.Commerce.getPlaySubscriptionStatus'), (_obj = {
  validatePlayPurchaseSync: _utils["default"]["synchronized"](_validatePlayPurchase.validatePlayPurchase),
  validatePlayPurchase: _utils["default"].promisified(_validatePlayPurchase.validatePlayPurchase),
  cancelPlaySubscriptionSync: _utils["default"]["synchronized"](_cancelPlaySubscription.cancelPlaySubscription),
  cancelPlaySubscription: _utils["default"].promisified(_cancelPlaySubscription.cancelPlaySubscription),
  getPlaySubscriptionStatusSync: _utils["default"]["synchronized"](_getPlaySubscriptionStatus.getPlaySubscriptionStatus),
  getPlaySubscriptionStatus: _utils["default"].promisified(_getPlaySubscriptionStatus.getPlaySubscriptionStatus)
}, ((0, _applyDecoratedDescriptor2["default"])(_obj, "validatePlayPurchaseSync", [_dec], (_init = Object.getOwnPropertyDescriptor(_obj, "validatePlayPurchaseSync"), _init = _init ? _init.value : undefined, {
  enumerable: true,
  configurable: true,
  writable: true,
  initializer: function initializer() {
    return _init;
  }
}), _obj), (0, _applyDecoratedDescriptor2["default"])(_obj, "cancelPlaySubscriptionSync", [_dec2], (_init2 = Object.getOwnPropertyDescriptor(_obj, "cancelPlaySubscriptionSync"), _init2 = _init2 ? _init2.value : undefined, {
  enumerable: true,
  configurable: true,
  writable: true,
  initializer: function initializer() {
    return _init2;
  }
}), _obj), (0, _applyDecoratedDescriptor2["default"])(_obj, "getPlaySubscriptionStatusSync", [_dec3], (_init3 = Object.getOwnPropertyDescriptor(_obj, "getPlaySubscriptionStatusSync"), _init3 = _init3 ? _init3.value : undefined, {
  enumerable: true,
  configurable: true,
  writable: true,
  initializer: function initializer() {
    return _init3;
  }
}), _obj)), _obj)));
var _default = Commerce;
exports["default"] = _default;
