// backendless/lib/counters/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _applyDecoratedDescriptor2 = _interopRequireDefault(require("@babel/runtime/helpers/applyDecoratedDescriptor"));

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _utils = _interopRequireDefault(require("../utils"));

var _decorators = require("../decorators");

var _counter = _interopRequireDefault(require("./counter"));

var _methods = require("./methods");

var _dec, _dec2, _dec3, _dec4, _dec5, _dec6, _dec7, _dec8, _dec9, _obj, _init, _init2, _init3, _init4, _init5, _init6, _init7, _init8, _init9;

var Counters = /*#__PURE__*/function () {
  function Counters(app) {
    (0, _classCallCheck2["default"])(this, Counters);
    this.app = app;
  }

  (0, _createClass2["default"])(Counters, [{
    key: "of",
    value: function of(name) {
      return new _counter["default"](name, this.app);
    }
  }]);
  return Counters;
}();

(0, _extends2["default"])(Counters.prototype, (_dec = (0, _decorators.deprecated)('Backendless.Counters', 'Backendless.Counters.incrementAndGet'), _dec2 = (0, _decorators.deprecated)('Backendless.Counters', 'Backendless.Counters.getAndIncrement'), _dec3 = (0, _decorators.deprecated)('Backendless.Counters', 'Backendless.Counters.decrementAndGet'), _dec4 = (0, _decorators.deprecated)('Backendless.Counters', 'Backendless.Counters.getAndDecrement'), _dec5 = (0, _decorators.deprecated)('Backendless.Counters', 'Backendless.Counters.reset'), _dec6 = (0, _decorators.deprecated)('Backendless.Counters', 'Backendless.Counters.get'), _dec7 = (0, _decorators.deprecated)('Backendless.Counters', 'Backendless.Counters.addAndGet'), _dec8 = (0, _decorators.deprecated)('Backendless.Counters', 'Backendless.Counters.getAndAdd'), _dec9 = (0, _decorators.deprecated)('Backendless.Counters', 'Backendless.Counters.compareAndSet'), (_obj = {
  incrementAndGetSync: _utils["default"]["synchronized"](_methods.incrementAndGet),
  incrementAndGet: _utils["default"].promisified(_methods.incrementAndGet),
  getAndIncrementSync: _utils["default"]["synchronized"](_methods.getAndIncrement),
  getAndIncrement: _utils["default"].promisified(_methods.getAndIncrement),
  decrementAndGetSync: _utils["default"]["synchronized"](_methods.decrementAndGet),
  decrementAndGet: _utils["default"].promisified(_methods.decrementAndGet),
  getAndDecrementSync: _utils["default"]["synchronized"](_methods.getAndDecrement),
  getAndDecrement: _utils["default"].promisified(_methods.getAndDecrement),
  resetSync: _utils["default"]["synchronized"](_methods.reset),
  reset: _utils["default"].promisified(_methods.reset),
  getSync: _utils["default"]["synchronized"](_methods.get),
  get: _utils["default"].promisified(_methods.get),
  addAndGetSync: _utils["default"]["synchronized"](_methods.addAndGet),
  addAndGet: _utils["default"].promisified(_methods.addAndGet),
  getAndAddSync: _utils["default"]["synchronized"](_methods.getAndAdd),
  getAndAdd: _utils["default"].promisified(_methods.getAndAdd),
  compareAndSetSync: _utils["default"]["synchronized"](_methods.compareAndSet),
  compareAndSet: _utils["default"].promisified(_methods.compareAndSet)
}, ((0, _applyDecoratedDescriptor2["default"])(_obj, "incrementAndGetSync", [_dec], (_init = Object.getOwnPropertyDescriptor(_obj, "incrementAndGetSync"), _init = _init ? _init.value : undefined, {
  enumerable: true,
  configurable: true,
  writable: true,
  initializer: function initializer() {
    return _init;
  }
}), _obj), (0, _applyDecoratedDescriptor2["default"])(_obj, "getAndIncrementSync", [_dec2], (_init2 = Object.getOwnPropertyDescriptor(_obj, "getAndIncrementSync"), _init2 = _init2 ? _init2.value : undefined, {
  enumerable: true,
  configurable: true,
  writable: true,
  initializer: function initializer() {
    return _init2;
  }
}), _obj), (0, _applyDecoratedDescriptor2["default"])(_obj, "decrementAndGetSync", [_dec3], (_init3 = Object.getOwnPropertyDescriptor(_obj, "decrementAndGetSync"), _init3 = _init3 ? _init3.value : undefined, {
  enumerable: true,
  configurable: true,
  writable: true,
  initializer: function initializer() {
    return _init3;
  }
}), _obj), (0, _applyDecoratedDescriptor2["default"])(_obj, "getAndDecrementSync", [_dec4], (_init4 = Object.getOwnPropertyDescriptor(_obj, "getAndDecrementSync"), _init4 = _init4 ? _init4.value : undefined, {
  enumerable: true,
  configurable: true,
  writable: true,
  initializer: function initializer() {
    return _init4;
  }
}), _obj), (0, _applyDecoratedDescriptor2["default"])(_obj, "resetSync", [_dec5], (_init5 = Object.getOwnPropertyDescriptor(_obj, "resetSync"), _init5 = _init5 ? _init5.value : undefined, {
  enumerable: true,
  configurable: true,
  writable: true,
  initializer: function initializer() {
    return _init5;
  }
}), _obj), (0, _applyDecoratedDescriptor2["default"])(_obj, "getSync", [_dec6], (_init6 = Object.getOwnPropertyDescriptor(_obj, "getSync"), _init6 = _init6 ? _init6.value : undefined, {
  enumerable: true,
  configurable: true,
  writable: true,
  initializer: function initializer() {
    return _init6;
  }
}), _obj), (0, _applyDecoratedDescriptor2["default"])(_obj, "addAndGetSync", [_dec7], (_init7 = Object.getOwnPropertyDescriptor(_obj, "addAndGetSync"), _init7 = _init7 ? _init7.value : undefined, {
  enumerable: true,
  configurable: true,
  writable: true,
  initializer: function initializer() {
    return _init7;
  }
}), _obj), (0, _applyDecoratedDescriptor2["default"])(_obj, "getAndAddSync", [_dec8], (_init8 = Object.getOwnPropertyDescriptor(_obj, "getAndAddSync"), _init8 = _init8 ? _init8.value : undefined, {
  enumerable: true,
  configurable: true,
  writable: true,
  initializer: function initializer() {
    return _init8;
  }
}), _obj), (0, _applyDecoratedDescriptor2["default"])(_obj, "compareAndSetSync", [_dec9], (_init9 = Object.getOwnPropertyDescriptor(_obj, "compareAndSetSync"), _init9 = _init9 ? _init9.value : undefined, {
  enumerable: true,
  configurable: true,
  writable: true,
  initializer: function initializer() {
    return _init9;
  }
}), _obj)), _obj)));
var _default = Counters;
exports["default"] = _default;
