// backendless/es/counters/counter.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _applyDecoratedDescriptor2 = _interopRequireDefault(require("@babel/runtime/helpers/applyDecoratedDescriptor"));

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _utils = _interopRequireDefault(require("../utils"));

var _decorators = require("../decorators");

var _methods = require("./methods");

var _dec, _dec2, _dec3, _dec4, _dec5, _dec6, _dec7, _dec8, _dec9, _obj, _init, _init2, _init3, _init4, _init5, _init6, _init7, _init8, _init9;

var Counter = function Counter(name, app) {
  (0, _classCallCheck2["default"])(this, Counter);

  if (!name || !_utils["default"].isString(name)) {
    throw new Error('Counter Name must be non empty String');
  }

  this.name = name;
  this.app = app;
};

var withCounterName = function withCounterName(method) {
  return function () {
    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    return method.call.apply(method, [this, this.name].concat(args));
  };
}; //TODO: will be removed when remove sync methods


var namespaceLabel = 'Backendless.Counter.of(<CounterName>)';
(0, _extends2["default"])(Counter.prototype, (_dec = (0, _decorators.deprecated)(namespaceLabel, "".concat(namespaceLabel, ".incrementAndGet")), _dec2 = (0, _decorators.deprecated)(namespaceLabel, "".concat(namespaceLabel, ".getAndIncrement")), _dec3 = (0, _decorators.deprecated)(namespaceLabel, "".concat(namespaceLabel, ".decrementAndGet")), _dec4 = (0, _decorators.deprecated)(namespaceLabel, "".concat(namespaceLabel, ".getAndDecrement")), _dec5 = (0, _decorators.deprecated)(namespaceLabel, "".concat(namespaceLabel, ".reset")), _dec6 = (0, _decorators.deprecated)(namespaceLabel, "".concat(namespaceLabel, ".get")), _dec7 = (0, _decorators.deprecated)(namespaceLabel, "".concat(namespaceLabel, ".addAndGet")), _dec8 = (0, _decorators.deprecated)(namespaceLabel, "".concat(namespaceLabel, ".getAndAdd")), _dec9 = (0, _decorators.deprecated)(namespaceLabel, "".concat(namespaceLabel, ".compareAndSet")), (_obj = {
  incrementAndGetSync: _utils["default"]["synchronized"](withCounterName(_methods.incrementAndGet)),
  incrementAndGet: _utils["default"].promisified(withCounterName(_methods.incrementAndGet)),
  getAndIncrementSync: _utils["default"]["synchronized"](withCounterName(_methods.getAndIncrement)),
  getAndIncrement: _utils["default"].promisified(withCounterName(_methods.getAndIncrement)),
  decrementAndGetSync: _utils["default"]["synchronized"](withCounterName(_methods.decrementAndGet)),
  decrementAndGet: _utils["default"].promisified(withCounterName(_methods.decrementAndGet)),
  getAndDecrementSync: _utils["default"]["synchronized"](withCounterName(_methods.getAndDecrement)),
  getAndDecrement: _utils["default"].promisified(withCounterName(_methods.getAndDecrement)),
  resetSync: _utils["default"]["synchronized"](withCounterName(_methods.reset)),
  reset: _utils["default"].promisified(withCounterName(_methods.reset)),
  getSync: _utils["default"]["synchronized"](withCounterName(_methods.get)),
  get: _utils["default"].promisified(withCounterName(_methods.get)),
  addAndGetSync: _utils["default"]["synchronized"](withCounterName(_methods.addAndGet)),
  addAndGet: _utils["default"].promisified(withCounterName(_methods.addAndGet)),
  getAndAddSync: _utils["default"]["synchronized"](withCounterName(_methods.getAndAdd)),
  getAndAdd: _utils["default"].promisified(withCounterName(_methods.getAndAdd)),
  compareAndSetSync: _utils["default"]["synchronized"](withCounterName(_methods.compareAndSet)),
  compareAndSet: _utils["default"].promisified(withCounterName(_methods.compareAndSet))
}, ((0, _applyDecoratedDescriptor2["default"])(_obj, "incrementAndGetSync", [_dec], (_init = Object.getOwnPropertyDescriptor(_obj, "incrementAndGetSync"), _init = _init ? _init.value : undefined, {
  enumerable: true,
  configurable: true,
  writable: true,
  initializer: function initializer() {
    return _init;
  }
}), _obj), (0, _applyDecoratedDescriptor2["default"])(_obj, "getAndIncrementSync", [_dec2], (_init2 = Object.getOwnPropertyDescriptor(_obj, "getAndIncrementSync"), _init2 = _init2 ? _init2.value : undefined, {
  enumerable: true,
  configurable: true,
  writable: true,
  initializer: function initializer() {
    return _init2;
  }
}), _obj), (0, _applyDecoratedDescriptor2["default"])(_obj, "decrementAndGetSync", [_dec3], (_init3 = Object.getOwnPropertyDescriptor(_obj, "decrementAndGetSync"), _init3 = _init3 ? _init3.value : undefined, {
  enumerable: true,
  configurable: true,
  writable: true,
  initializer: function initializer() {
    return _init3;
  }
}), _obj), (0, _applyDecoratedDescriptor2["default"])(_obj, "getAndDecrementSync", [_dec4], (_init4 = Object.getOwnPropertyDescriptor(_obj, "getAndDecrementSync"), _init4 = _init4 ? _init4.value : undefined, {
  enumerable: true,
  configurable: true,
  writable: true,
  initializer: function initializer() {
    return _init4;
  }
}), _obj), (0, _applyDecoratedDescriptor2["default"])(_obj, "resetSync", [_dec5], (_init5 = Object.getOwnPropertyDescriptor(_obj, "resetSync"), _init5 = _init5 ? _init5.value : undefined, {
  enumerable: true,
  configurable: true,
  writable: true,
  initializer: function initializer() {
    return _init5;
  }
}), _obj), (0, _applyDecoratedDescriptor2["default"])(_obj, "getSync", [_dec6], (_init6 = Object.getOwnPropertyDescriptor(_obj, "getSync"), _init6 = _init6 ? _init6.value : undefined, {
  enumerable: true,
  configurable: true,
  writable: true,
  initializer: function initializer() {
    return _init6;
  }
}), _obj), (0, _applyDecoratedDescriptor2["default"])(_obj, "addAndGetSync", [_dec7], (_init7 = Object.getOwnPropertyDescriptor(_obj, "addAndGetSync"), _init7 = _init7 ? _init7.value : undefined, {
  enumerable: true,
  configurable: true,
  writable: true,
  initializer: function initializer() {
    return _init7;
  }
}), _obj), (0, _applyDecoratedDescriptor2["default"])(_obj, "getAndAddSync", [_dec8], (_init8 = Object.getOwnPropertyDescriptor(_obj, "getAndAddSync"), _init8 = _init8 ? _init8.value : undefined, {
  enumerable: true,
  configurable: true,
  writable: true,
  initializer: function initializer() {
    return _init8;
  }
}), _obj), (0, _applyDecoratedDescriptor2["default"])(_obj, "compareAndSetSync", [_dec9], (_init9 = Object.getOwnPropertyDescriptor(_obj, "compareAndSetSync"), _init9 = _init9 ? _init9.value : undefined, {
  enumerable: true,
  configurable: true,
  writable: true,
  initializer: function initializer() {
    return _init9;
  }
}), _obj)), _obj)));
var _default = Counter;
exports["default"] = _default;
