// backendless/es/messaging/methods/publish.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.publish = publish;

var _utils = _interopRequireDefault(require("../../utils"));

var _async = _interopRequireDefault(require("../../request/async"));

var _deliveryOptions = _interopRequireDefault(require("../helpers/delivery-options"));

var _publishOptions = _interopRequireDefault(require("../helpers/publish-options"));

function publish(channelName, message, publishOptions, deliveryTarget, asyncHandler) {
  if (publishOptions instanceof _async["default"]) {
    asyncHandler = publishOptions;
    publishOptions = undefined;
    deliveryTarget = undefined;
  }

  if (deliveryTarget instanceof _async["default"]) {
    asyncHandler = deliveryTarget;
    deliveryTarget = undefined;
  }

  var data = {
    message: message
  };

  if (publishOptions) {
    if (!(publishOptions instanceof _publishOptions["default"])) {
      throw new Error('Use PublishOption as publishOptions argument');
    }

    _utils["default"].deepExtend(data, publishOptions);
  }

  if (deliveryTarget) {
    if (!(deliveryTarget instanceof _deliveryOptions["default"])) {
      throw new Error('Use DeliveryOptions as deliveryTarget argument');
    }

    _utils["default"].deepExtend(data, deliveryTarget);
  }

  return this.app.request.post({
    url: this.app.urls.messagingChannel(channelName),
    isAsync: !!asyncHandler,
    asyncHandler: asyncHandler,
    data: data
  });
}
