// backendless/es/messaging/methods/register-device.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.registerDevice = registerDevice;

var _utils = _interopRequireDefault(require("../../utils"));

var _async = _interopRequireDefault(require("../../request/async"));

function registerDevice(deviceToken, channels, expiration, asyncHandler) {
  var device = this.app.device;
  asyncHandler = _utils["default"].extractResponder(arguments);
  var data = {
    deviceToken: deviceToken,
    deviceId: device.uuid,
    os: device.platform,
    osVersion: device.version
  };

  if (_utils["default"].isArray(channels)) {
    data.channels = channels;
  }

  if (_utils["default"].isNumber(expiration) || expiration instanceof Date) {
    data.expiration = expiration instanceof Date ? expiration.getTime() / 1000 : expiration;
  }

  return this.app.request.post({
    url: this.app.urls.messagingRegistrations(),
    data: data,
    isAsync: !!asyncHandler,
    asyncHandler: asyncHandler
  });
}
