// backendless/es/geo/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _applyDecoratedDescriptor2 = _interopRequireDefault(require("@babel/runtime/helpers/applyDecoratedDescriptor"));

var _utils = _interopRequireDefault(require("../utils"));

var _utils2 = _interopRequireDefault(require("./utils"));

var _decorators = require("../decorators");

var _constants = require("./constants");

var _trackerMonitor = _interopRequireDefault(require("./tracker-monitor"));

var _cluster = _interopRequireDefault(require("./cluster"));

var _point = _interopRequireDefault(require("./point"));

var _query = _interopRequireDefault(require("./query"));

var _savePoint2 = require("./save-point");

var _addCategory2 = require("./add-category");

var _deleteCategory2 = require("./delete-category");

var _getCategories2 = require("./get-categories");

var _deletePoint2 = require("./delete-point");

var _getPointsCount = require("./get-points-count");

var _getClusterPoints2 = require("./get-cluster-points");

var _loadPoints = require("./load-points");

var _loadMetadata2 = require("./load-metadata");

var _relativeFind2 = require("./relative-find");

var _loadFencePoints = require("./load-fence-points");

var _dec, _dec2, _dec3, _dec4, _dec5, _dec6, _dec7, _dec8, _dec9, _dec10, _dec11, _dec12, _dec13, _dec14, _dec15, _dec16, _dec17, _dec18, _dec19, _dec20, _dec21, _dec22, _dec23, _dec24, _dec25, _dec26, _dec27, _dec28, _dec29, _dec30, _dec31, _dec32, _dec33, _dec34, _class;

var alternativeForDepreciation = 'Spatial Data Types [POINT,LINESTRING,POLYGON]. ' + 'See more details here: https://backendless.com/docs/js/data_spatial_overview.html ';
var Geo = (_dec = (0, _decorators.deprecated)('Backendless.Geo', alternativeForDepreciation), _dec2 = (0, _decorators.deprecated)('Backendless.Geo', alternativeForDepreciation), _dec3 = (0, _decorators.deprecated)('Backendless.Geo', alternativeForDepreciation), _dec4 = (0, _decorators.deprecated)('Backendless.Geo', alternativeForDepreciation), _dec5 = (0, _decorators.deprecated)('Backendless.Geo', alternativeForDepreciation), _dec6 = (0, _decorators.deprecated)('Backendless.Geo', alternativeForDepreciation), _dec7 = (0, _decorators.deprecated)('Backendless.Geo', alternativeForDepreciation), _dec8 = (0, _decorators.deprecated)('Backendless.Geo', alternativeForDepreciation), _dec9 = (0, _decorators.deprecated)('Backendless.Geo', alternativeForDepreciation), _dec10 = (0, _decorators.deprecated)('Backendless.Geo', alternativeForDepreciation), _dec11 = (0, _decorators.deprecated)('Backendless.Geo', alternativeForDepreciation), _dec12 = (0, _decorators.deprecated)('Backendless.Geo', alternativeForDepreciation), _dec13 = (0, _decorators.deprecated)('Backendless.Geo', alternativeForDepreciation), _dec14 = (0, _decorators.deprecated)('Backendless.Geo', alternativeForDepreciation), _dec15 = (0, _decorators.deprecated)('Backendless.Geo', alternativeForDepreciation), _dec16 = (0, _decorators.deprecated)('Backendless.Geo', alternativeForDepreciation), _dec17 = (0, _decorators.deprecated)('Backendless.Geo', alternativeForDepreciation), _dec18 = (0, _decorators.deprecated)('Backendless.Geo', alternativeForDepreciation), _dec19 = (0, _decorators.deprecated)('Backendless.Geo', alternativeForDepreciation), _dec20 = (0, _decorators.deprecated)('Backendless.Geo', alternativeForDepreciation), _dec21 = (0, _decorators.deprecated)('Backendless.Geo', alternativeForDepreciation), _dec22 = (0, _decorators.deprecated)('Backendless.Geo', alternativeForDepreciation), _dec23 = (0, _decorators.deprecated)('Backendless.Geo', alternativeForDepreciation), _dec24 = (0, _decorators.deprecated)('Backendless.Geo'), _dec25 = (0, _decorators.deprecated)('Backendless.Geo'), _dec26 = (0, _decorators.deprecated)('Backendless.Geo'), _dec27 = (0, _decorators.deprecated)('Backendless.Geo'), _dec28 = (0, _decorators.deprecated)('Backendless.Geo'), _dec29 = (0, _decorators.deprecated)('Backendless.Geo'), _dec30 = (0, _decorators.deprecated)('Backendless.Geo'), _dec31 = (0, _decorators.deprecated)('Backendless.Geo'), _dec32 = (0, _decorators.deprecated)('Backendless.Geo'), _dec33 = (0, _decorators.deprecated)('Backendless.Geo'), _dec34 = (0, _decorators.deprecated)('Backendless.Geo'), (_class = /*#__PURE__*/function () {
  function Geo(app) {
    (0, _classCallCheck2["default"])(this, Geo);
    this.app = app;
    this.trackerMonitor = new _trackerMonitor["default"](app);
    this.EARTH_RADIUS = _constants.EARTH_RADIUS;
    this.UNITS = _constants.UNITS;
    this.Utils = _utils2["default"];
    this.Cluster = _cluster["default"];
    this.Point = _point["default"];
    this.Query = _query["default"];
  }

  (0, _createClass2["default"])(Geo, [{
    key: "addPoint",
    value: function addPoint()
    /** geopoint, async */
    {
      return this.savePoint.apply(this, arguments);
    }
  }, {
    key: "savePointSync",
    value: function savePointSync() {
      var _Utils$synchronized;

      for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
        args[_key] = arguments[_key];
      }

      return (_Utils$synchronized = _utils["default"]["synchronized"](_savePoint2.savePoint)).call.apply(_Utils$synchronized, [this].concat(args));
    }
  }, {
    key: "savePoint",
    value: function savePoint() {
      var _Utils$promisified;

      for (var _len2 = arguments.length, args = new Array(_len2), _key2 = 0; _key2 < _len2; _key2++) {
        args[_key2] = arguments[_key2];
      }

      return (_Utils$promisified = _utils["default"].promisified(_savePoint2.savePoint)).call.apply(_Utils$promisified, [this].concat(args));
    }
  }, {
    key: "findSync",
    value: function findSync() {
      var _Utils$synchronized2;

      for (var _len3 = arguments.length, args = new Array(_len3), _key3 = 0; _key3 < _len3; _key3++) {
        args[_key3] = arguments[_key3];
      }

      return (_Utils$synchronized2 = _utils["default"]["synchronized"](_loadPoints.loadPoints)).call.apply(_Utils$synchronized2, [this].concat(args));
    }
  }, {
    key: "find",
    value: function find() {
      var _Utils$promisified2;

      for (var _len4 = arguments.length, args = new Array(_len4), _key4 = 0; _key4 < _len4; _key4++) {
        args[_key4] = arguments[_key4];
      }

      return (_Utils$promisified2 = _utils["default"].promisified(_loadPoints.loadPoints)).call.apply(_Utils$promisified2, [this].concat(args));
    }
  }, {
    key: "loadMetadataSync",
    value: function loadMetadataSync() {
      var _Utils$synchronized3;

      for (var _len5 = arguments.length, args = new Array(_len5), _key5 = 0; _key5 < _len5; _key5++) {
        args[_key5] = arguments[_key5];
      }

      return (_Utils$synchronized3 = _utils["default"]["synchronized"](_loadMetadata2.loadMetadata)).call.apply(_Utils$synchronized3, [this].concat(args));
    }
  }, {
    key: "loadMetadata",
    value: function loadMetadata() {
      var _Utils$promisified3;

      for (var _len6 = arguments.length, args = new Array(_len6), _key6 = 0; _key6 < _len6; _key6++) {
        args[_key6] = arguments[_key6];
      }

      return (_Utils$promisified3 = _utils["default"].promisified(_loadMetadata2.loadMetadata)).call.apply(_Utils$promisified3, [this].concat(args));
    }
  }, {
    key: "getClusterPointsSync",
    value: function getClusterPointsSync() {
      var _Utils$synchronized4;

      for (var _len7 = arguments.length, args = new Array(_len7), _key7 = 0; _key7 < _len7; _key7++) {
        args[_key7] = arguments[_key7];
      }

      return (_Utils$synchronized4 = _utils["default"]["synchronized"](_getClusterPoints2.getClusterPoints)).call.apply(_Utils$synchronized4, [this].concat(args));
    }
  }, {
    key: "getClusterPoints",
    value: function getClusterPoints() {
      var _Utils$promisified4;

      for (var _len8 = arguments.length, args = new Array(_len8), _key8 = 0; _key8 < _len8; _key8++) {
        args[_key8] = arguments[_key8];
      }

      return (_Utils$promisified4 = _utils["default"].promisified(_getClusterPoints2.getClusterPoints)).call.apply(_Utils$promisified4, [this].concat(args));
    }
  }, {
    key: "relativeFindSync",
    value: function relativeFindSync() {
      var _Utils$synchronized5;

      for (var _len9 = arguments.length, args = new Array(_len9), _key9 = 0; _key9 < _len9; _key9++) {
        args[_key9] = arguments[_key9];
      }

      return (_Utils$synchronized5 = _utils["default"]["synchronized"](_relativeFind2.relativeFind)).call.apply(_Utils$synchronized5, [this].concat(args));
    }
  }, {
    key: "relativeFind",
    value: function relativeFind() {
      var _Utils$promisified5;

      for (var _len10 = arguments.length, args = new Array(_len10), _key10 = 0; _key10 < _len10; _key10++) {
        args[_key10] = arguments[_key10];
      }

      return (_Utils$promisified5 = _utils["default"].promisified(_relativeFind2.relativeFind)).call.apply(_Utils$promisified5, [this].concat(args));
    }
  }, {
    key: "addCategorySync",
    value: function addCategorySync() {
      var _Utils$synchronized6;

      for (var _len11 = arguments.length, args = new Array(_len11), _key11 = 0; _key11 < _len11; _key11++) {
        args[_key11] = arguments[_key11];
      }

      return (_Utils$synchronized6 = _utils["default"]["synchronized"](_addCategory2.addCategory)).call.apply(_Utils$synchronized6, [this].concat(args));
    }
  }, {
    key: "addCategory",
    value: function addCategory() {
      var _Utils$promisified6;

      for (var _len12 = arguments.length, args = new Array(_len12), _key12 = 0; _key12 < _len12; _key12++) {
        args[_key12] = arguments[_key12];
      }

      return (_Utils$promisified6 = _utils["default"].promisified(_addCategory2.addCategory)).call.apply(_Utils$promisified6, [this].concat(args));
    }
  }, {
    key: "getCategoriesSync",
    value: function getCategoriesSync() {
      var _Utils$synchronized7;

      for (var _len13 = arguments.length, args = new Array(_len13), _key13 = 0; _key13 < _len13; _key13++) {
        args[_key13] = arguments[_key13];
      }

      return (_Utils$synchronized7 = _utils["default"]["synchronized"](_getCategories2.getCategories)).call.apply(_Utils$synchronized7, [this].concat(args));
    }
  }, {
    key: "getCategories",
    value: function getCategories() {
      var _Utils$promisified7;

      for (var _len14 = arguments.length, args = new Array(_len14), _key14 = 0; _key14 < _len14; _key14++) {
        args[_key14] = arguments[_key14];
      }

      return (_Utils$promisified7 = _utils["default"].promisified(_getCategories2.getCategories)).call.apply(_Utils$promisified7, [this].concat(args));
    }
  }, {
    key: "deleteCategorySync",
    value: function deleteCategorySync() {
      var _Utils$synchronized8;

      for (var _len15 = arguments.length, args = new Array(_len15), _key15 = 0; _key15 < _len15; _key15++) {
        args[_key15] = arguments[_key15];
      }

      return (_Utils$synchronized8 = _utils["default"]["synchronized"](_deleteCategory2.deleteCategory)).call.apply(_Utils$synchronized8, [this].concat(args));
    }
  }, {
    key: "deleteCategory",
    value: function deleteCategory() {
      var _Utils$promisified8;

      for (var _len16 = arguments.length, args = new Array(_len16), _key16 = 0; _key16 < _len16; _key16++) {
        args[_key16] = arguments[_key16];
      }

      return (_Utils$promisified8 = _utils["default"].promisified(_deleteCategory2.deleteCategory)).call.apply(_Utils$promisified8, [this].concat(args));
    }
  }, {
    key: "deletePointSync",
    value: function deletePointSync() {
      var _Utils$synchronized9;

      for (var _len17 = arguments.length, args = new Array(_len17), _key17 = 0; _key17 < _len17; _key17++) {
        args[_key17] = arguments[_key17];
      }

      return (_Utils$synchronized9 = _utils["default"]["synchronized"](_deletePoint2.deletePoint)).call.apply(_Utils$synchronized9, [this].concat(args));
    }
  }, {
    key: "deletePoint",
    value: function deletePoint() {
      var _Utils$promisified9;

      for (var _len18 = arguments.length, args = new Array(_len18), _key18 = 0; _key18 < _len18; _key18++) {
        args[_key18] = arguments[_key18];
      }

      return (_Utils$promisified9 = _utils["default"].promisified(_deletePoint2.deletePoint)).call.apply(_Utils$promisified9, [this].concat(args));
    }
  }, {
    key: "getFencePointsSync",
    value: function getFencePointsSync() {
      var _Utils$synchronized10;

      for (var _len19 = arguments.length, args = new Array(_len19), _key19 = 0; _key19 < _len19; _key19++) {
        args[_key19] = arguments[_key19];
      }

      return (_Utils$synchronized10 = _utils["default"]["synchronized"](_loadFencePoints.loadFencePoints)).call.apply(_Utils$synchronized10, [this].concat(args));
    }
  }, {
    key: "getFencePoints",
    value: function getFencePoints() {
      var _Utils$promisified10;

      for (var _len20 = arguments.length, args = new Array(_len20), _key20 = 0; _key20 < _len20; _key20++) {
        args[_key20] = arguments[_key20];
      }

      return (_Utils$promisified10 = _utils["default"].promisified(_loadFencePoints.loadFencePoints)).call.apply(_Utils$promisified10, [this].concat(args));
    }
  }, {
    key: "getGeopointCountSync",
    value: function getGeopointCountSync() {
      var _Utils$synchronized11;

      for (var _len21 = arguments.length, args = new Array(_len21), _key21 = 0; _key21 < _len21; _key21++) {
        args[_key21] = arguments[_key21];
      }

      return (_Utils$synchronized11 = _utils["default"]["synchronized"](_getPointsCount.getGeopointCount)).call.apply(_Utils$synchronized11, [this].concat(args));
    }
  }, {
    key: "getGeopointCount",
    value: function getGeopointCount() {
      var _Utils$promisified11;

      for (var _len22 = arguments.length, args = new Array(_len22), _key22 = 0; _key22 < _len22; _key22++) {
        args[_key22] = arguments[_key22];
      }

      return (_Utils$promisified11 = _utils["default"].promisified(_getPointsCount.getGeopointCount)).call.apply(_Utils$promisified11, [this].concat(args));
    }
  }, {
    key: "runOnEnterActionSync",
    value: function runOnEnterActionSync() {
      var _this$trackerMonitor;

      return (_this$trackerMonitor = this.trackerMonitor).runOnEnterActionSync.apply(_this$trackerMonitor, arguments);
    }
  }, {
    key: "runOnEnterAction",
    value: function runOnEnterAction() {
      var _this$trackerMonitor2;

      return (_this$trackerMonitor2 = this.trackerMonitor).runOnEnterAction.apply(_this$trackerMonitor2, arguments);
    }
  }, {
    key: "runOnStayActionSync",
    value: function runOnStayActionSync() {
      var _this$trackerMonitor3;

      return (_this$trackerMonitor3 = this.trackerMonitor).runOnStayActionSync.apply(_this$trackerMonitor3, arguments);
    }
  }, {
    key: "runOnStayAction",
    value: function runOnStayAction() {
      var _this$trackerMonitor4;

      return (_this$trackerMonitor4 = this.trackerMonitor).runOnStayAction.apply(_this$trackerMonitor4, arguments);
    }
  }, {
    key: "runOnExitActionSync",
    value: function runOnExitActionSync() {
      var _this$trackerMonitor5;

      return (_this$trackerMonitor5 = this.trackerMonitor).runOnExitActionSync.apply(_this$trackerMonitor5, arguments);
    }
  }, {
    key: "runOnExitAction",
    value: function runOnExitAction() {
      var _this$trackerMonitor6;

      return (_this$trackerMonitor6 = this.trackerMonitor).runOnExitAction.apply(_this$trackerMonitor6, arguments);
    }
  }, {
    key: "startGeofenceMonitoringWithInAppCallbackSync",
    value: function startGeofenceMonitoringWithInAppCallbackSync() {
      var _this$trackerMonitor7;

      return (_this$trackerMonitor7 = this.trackerMonitor).startGeofenceMonitoringWithInAppCallbackSync.apply(_this$trackerMonitor7, arguments);
    }
  }, {
    key: "startGeofenceMonitoringWithInAppCallback",
    value: function startGeofenceMonitoringWithInAppCallback() {
      var _this$trackerMonitor8;

      return (_this$trackerMonitor8 = this.trackerMonitor).startGeofenceMonitoringWithInAppCallback.apply(_this$trackerMonitor8, arguments);
    }
  }, {
    key: "startGeofenceMonitoringWithRemoteCallbackSync",
    value: function startGeofenceMonitoringWithRemoteCallbackSync() {
      var _this$trackerMonitor9;

      return (_this$trackerMonitor9 = this.trackerMonitor).startGeofenceMonitoringWithRemoteCallbackSync.apply(_this$trackerMonitor9, arguments);
    }
  }, {
    key: "startGeofenceMonitoringWithRemoteCallback",
    value: function startGeofenceMonitoringWithRemoteCallback() {
      var _this$trackerMonitor10;

      return (_this$trackerMonitor10 = this.trackerMonitor).startGeofenceMonitoringWithRemoteCallback.apply(_this$trackerMonitor10, arguments);
    }
  }, {
    key: "stopGeofenceMonitoring",
    value: function stopGeofenceMonitoring() {
      var _this$trackerMonitor11;

      return (_this$trackerMonitor11 = this.trackerMonitor).stopGeofenceMonitoring.apply(_this$trackerMonitor11, arguments);
    }
  }, {
    key: "resetGeofenceMonitoring",
    value: function resetGeofenceMonitoring() {
      var _this$trackerMonitor12;

      return (_this$trackerMonitor12 = this.trackerMonitor).reset.apply(_this$trackerMonitor12, arguments);
    }
  }]);
  return Geo;
}(), ((0, _applyDecoratedDescriptor2["default"])(_class.prototype, "addPoint", [_dec], Object.getOwnPropertyDescriptor(_class.prototype, "addPoint"), _class.prototype), (0, _applyDecoratedDescriptor2["default"])(_class.prototype, "savePointSync", [_dec2], Object.getOwnPropertyDescriptor(_class.prototype, "savePointSync"), _class.prototype), (0, _applyDecoratedDescriptor2["default"])(_class.prototype, "savePoint", [_dec3], Object.getOwnPropertyDescriptor(_class.prototype, "savePoint"), _class.prototype), (0, _applyDecoratedDescriptor2["default"])(_class.prototype, "findSync", [_dec4], Object.getOwnPropertyDescriptor(_class.prototype, "findSync"), _class.prototype), (0, _applyDecoratedDescriptor2["default"])(_class.prototype, "find", [_dec5], Object.getOwnPropertyDescriptor(_class.prototype, "find"), _class.prototype), (0, _applyDecoratedDescriptor2["default"])(_class.prototype, "loadMetadataSync", [_dec6], Object.getOwnPropertyDescriptor(_class.prototype, "loadMetadataSync"), _class.prototype), (0, _applyDecoratedDescriptor2["default"])(_class.prototype, "loadMetadata", [_dec7], Object.getOwnPropertyDescriptor(_class.prototype, "loadMetadata"), _class.prototype), (0, _applyDecoratedDescriptor2["default"])(_class.prototype, "getClusterPointsSync", [_dec8], Object.getOwnPropertyDescriptor(_class.prototype, "getClusterPointsSync"), _class.prototype), (0, _applyDecoratedDescriptor2["default"])(_class.prototype, "getClusterPoints", [_dec9], Object.getOwnPropertyDescriptor(_class.prototype, "getClusterPoints"), _class.prototype), (0, _applyDecoratedDescriptor2["default"])(_class.prototype, "relativeFindSync", [_dec10], Object.getOwnPropertyDescriptor(_class.prototype, "relativeFindSync"), _class.prototype), (0, _applyDecoratedDescriptor2["default"])(_class.prototype, "relativeFind", [_dec11], Object.getOwnPropertyDescriptor(_class.prototype, "relativeFind"), _class.prototype), (0, _applyDecoratedDescriptor2["default"])(_class.prototype, "addCategorySync", [_dec12], Object.getOwnPropertyDescriptor(_class.prototype, "addCategorySync"), _class.prototype), (0, _applyDecoratedDescriptor2["default"])(_class.prototype, "addCategory", [_dec13], Object.getOwnPropertyDescriptor(_class.prototype, "addCategory"), _class.prototype), (0, _applyDecoratedDescriptor2["default"])(_class.prototype, "getCategoriesSync", [_dec14], Object.getOwnPropertyDescriptor(_class.prototype, "getCategoriesSync"), _class.prototype), (0, _applyDecoratedDescriptor2["default"])(_class.prototype, "getCategories", [_dec15], Object.getOwnPropertyDescriptor(_class.prototype, "getCategories"), _class.prototype), (0, _applyDecoratedDescriptor2["default"])(_class.prototype, "deleteCategorySync", [_dec16], Object.getOwnPropertyDescriptor(_class.prototype, "deleteCategorySync"), _class.prototype), (0, _applyDecoratedDescriptor2["default"])(_class.prototype, "deleteCategory", [_dec17], Object.getOwnPropertyDescriptor(_class.prototype, "deleteCategory"), _class.prototype), (0, _applyDecoratedDescriptor2["default"])(_class.prototype, "deletePointSync", [_dec18], Object.getOwnPropertyDescriptor(_class.prototype, "deletePointSync"), _class.prototype), (0, _applyDecoratedDescriptor2["default"])(_class.prototype, "deletePoint", [_dec19], Object.getOwnPropertyDescriptor(_class.prototype, "deletePoint"), _class.prototype), (0, _applyDecoratedDescriptor2["default"])(_class.prototype, "getFencePointsSync", [_dec20], Object.getOwnPropertyDescriptor(_class.prototype, "getFencePointsSync"), _class.prototype), (0, _applyDecoratedDescriptor2["default"])(_class.prototype, "getFencePoints", [_dec21], Object.getOwnPropertyDescriptor(_class.prototype, "getFencePoints"), _class.prototype), (0, _applyDecoratedDescriptor2["default"])(_class.prototype, "getGeopointCountSync", [_dec22], Object.getOwnPropertyDescriptor(_class.prototype, "getGeopointCountSync"), _class.prototype), (0, _applyDecoratedDescriptor2["default"])(_class.prototype, "getGeopointCount", [_dec23], Object.getOwnPropertyDescriptor(_class.prototype, "getGeopointCount"), _class.prototype), (0, _applyDecoratedDescriptor2["default"])(_class.prototype, "runOnEnterActionSync", [_dec24], Object.getOwnPropertyDescriptor(_class.prototype, "runOnEnterActionSync"), _class.prototype), (0, _applyDecoratedDescriptor2["default"])(_class.prototype, "runOnEnterAction", [_dec25], Object.getOwnPropertyDescriptor(_class.prototype, "runOnEnterAction"), _class.prototype), (0, _applyDecoratedDescriptor2["default"])(_class.prototype, "runOnStayActionSync", [_dec26], Object.getOwnPropertyDescriptor(_class.prototype, "runOnStayActionSync"), _class.prototype), (0, _applyDecoratedDescriptor2["default"])(_class.prototype, "runOnStayAction", [_dec27], Object.getOwnPropertyDescriptor(_class.prototype, "runOnStayAction"), _class.prototype), (0, _applyDecoratedDescriptor2["default"])(_class.prototype, "runOnExitActionSync", [_dec28], Object.getOwnPropertyDescriptor(_class.prototype, "runOnExitActionSync"), _class.prototype), (0, _applyDecoratedDescriptor2["default"])(_class.prototype, "runOnExitAction", [_dec29], Object.getOwnPropertyDescriptor(_class.prototype, "runOnExitAction"), _class.prototype), (0, _applyDecoratedDescriptor2["default"])(_class.prototype, "startGeofenceMonitoringWithInAppCallbackSync", [_dec30], Object.getOwnPropertyDescriptor(_class.prototype, "startGeofenceMonitoringWithInAppCallbackSync"), _class.prototype), (0, _applyDecoratedDescriptor2["default"])(_class.prototype, "startGeofenceMonitoringWithInAppCallback", [_dec31], Object.getOwnPropertyDescriptor(_class.prototype, "startGeofenceMonitoringWithInAppCallback"), _class.prototype), (0, _applyDecoratedDescriptor2["default"])(_class.prototype, "startGeofenceMonitoringWithRemoteCallbackSync", [_dec32], Object.getOwnPropertyDescriptor(_class.prototype, "startGeofenceMonitoringWithRemoteCallbackSync"), _class.prototype), (0, _applyDecoratedDescriptor2["default"])(_class.prototype, "startGeofenceMonitoringWithRemoteCallback", [_dec33], Object.getOwnPropertyDescriptor(_class.prototype, "startGeofenceMonitoringWithRemoteCallback"), _class.prototype), (0, _applyDecoratedDescriptor2["default"])(_class.prototype, "stopGeofenceMonitoring", [_dec34], Object.getOwnPropertyDescriptor(_class.prototype, "stopGeofenceMonitoring"), _class.prototype)), _class));
exports["default"] = Geo;
