// backendless/lib/cache/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _applyDecoratedDescriptor2 = _interopRequireDefault(require("@babel/runtime/helpers/applyDecoratedDescriptor"));

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _utils = _interopRequireDefault(require("../utils"));

var _decorators = require("../decorators");

var _put = require("./put");

var _get = require("./get");

var _remove = require("./remove");

var _contains = require("./contains");

var _expireIn = require("./expire-in");

var _expireAt = require("./expire-at");

var _parsers = require("./parsers");

var _dec, _dec2, _dec3, _dec4, _dec5, _dec6, _obj, _init, _init2, _init3, _init4, _init5, _init6;

var Cache = /*#__PURE__*/function () {
  function Cache(app) {
    (0, _classCallCheck2["default"])(this, Cache);
    this.app = app;
  } //TODO: do we need it?


  (0, _createClass2["default"])(Cache, [{
    key: "setObjectFactory",
    value: function setObjectFactory(objectName, factoryMethod) {
      _parsers.Parsers.set(objectName, factoryMethod);
    }
  }]);
  return Cache;
}();

(0, _extends2["default"])(Cache.prototype, (_dec = (0, _decorators.deprecated)('Backendless.Cache', 'Backendless.Cache.put'), _dec2 = (0, _decorators.deprecated)('Backendless.Cache', 'Backendless.Cache.get'), _dec3 = (0, _decorators.deprecated)('Backendless.Cache', 'Backendless.Cache.remove'), _dec4 = (0, _decorators.deprecated)('Backendless.Cache', 'Backendless.Cache.contains'), _dec5 = (0, _decorators.deprecated)('Backendless.Cache', 'Backendless.Cache.expireIn'), _dec6 = (0, _decorators.deprecated)('Backendless.Cache', 'Backendless.Cache.expireAt'), (_obj = {
  putSync: _utils["default"]["synchronized"](_put.put),
  put: _utils["default"].promisified(_put.put),
  getSync: _utils["default"]["synchronized"](_get.get),
  get: _utils["default"].promisified(_get.get),
  removeSync: _utils["default"]["synchronized"](_remove.remove),
  remove: _utils["default"].promisified(_remove.remove),
  containsSync: _utils["default"]["synchronized"](_contains.contains),
  contains: _utils["default"].promisified(_contains.contains),
  expireInSync: _utils["default"]["synchronized"](_expireIn.expireIn),
  expireIn: _utils["default"].promisified(_expireIn.expireIn),
  expireAtSync: _utils["default"]["synchronized"](_expireAt.expireAt),
  expireAt: _utils["default"].promisified(_expireAt.expireAt)
}, ((0, _applyDecoratedDescriptor2["default"])(_obj, "putSync", [_dec], (_init = Object.getOwnPropertyDescriptor(_obj, "putSync"), _init = _init ? _init.value : undefined, {
  enumerable: true,
  configurable: true,
  writable: true,
  initializer: function initializer() {
    return _init;
  }
}), _obj), (0, _applyDecoratedDescriptor2["default"])(_obj, "getSync", [_dec2], (_init2 = Object.getOwnPropertyDescriptor(_obj, "getSync"), _init2 = _init2 ? _init2.value : undefined, {
  enumerable: true,
  configurable: true,
  writable: true,
  initializer: function initializer() {
    return _init2;
  }
}), _obj), (0, _applyDecoratedDescriptor2["default"])(_obj, "removeSync", [_dec3], (_init3 = Object.getOwnPropertyDescriptor(_obj, "removeSync"), _init3 = _init3 ? _init3.value : undefined, {
  enumerable: true,
  configurable: true,
  writable: true,
  initializer: function initializer() {
    return _init3;
  }
}), _obj), (0, _applyDecoratedDescriptor2["default"])(_obj, "containsSync", [_dec4], (_init4 = Object.getOwnPropertyDescriptor(_obj, "containsSync"), _init4 = _init4 ? _init4.value : undefined, {
  enumerable: true,
  configurable: true,
  writable: true,
  initializer: function initializer() {
    return _init4;
  }
}), _obj), (0, _applyDecoratedDescriptor2["default"])(_obj, "expireInSync", [_dec5], (_init5 = Object.getOwnPropertyDescriptor(_obj, "expireInSync"), _init5 = _init5 ? _init5.value : undefined, {
  enumerable: true,
  configurable: true,
  writable: true,
  initializer: function initializer() {
    return _init5;
  }
}), _obj), (0, _applyDecoratedDescriptor2["default"])(_obj, "expireAtSync", [_dec6], (_init6 = Object.getOwnPropertyDescriptor(_obj, "expireAtSync"), _init6 = _init6 ? _init6.value : undefined, {
  enumerable: true,
  configurable: true,
  writable: true,
  initializer: function initializer() {
    return _init6;
  }
}), _obj)), _obj)));
var _default = Cache;
exports["default"] = _default;
