// backendless/es/counters/methods/compare-and-set.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.compareAndSet = compareAndSet;

var _utils = _interopRequireDefault(require("../../utils"));

function compareAndSet(counterName, expected, updated, asyncHandler) {
  if (!counterName || !_utils["default"].isString(counterName)) {
    throw new Error('Counter Name must be non empty String');
  }

  if (!_utils["default"].isNumber(expected)) {
    throw new Error('Counter "expected" value must be Number');
  }

  if (!_utils["default"].isNumber(updated)) {
    throw new Error('Counter "updated" value must be Number');
  }

  if (asyncHandler) {
    asyncHandler = _utils["default"].wrapAsync(asyncHandler);
  }

  return this.app.request.put({
    url: this.app.urls.counterCompareAndSet(counterName),
    query: {
      expected: expected,
      updatedvalue: updated
    },
    isAsync: !!asyncHandler,
    asyncHandler: asyncHandler
  });
}
