// backendless/es/logging/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _applyDecoratedDescriptor2 = _interopRequireDefault(require("@babel/runtime/helpers/applyDecoratedDescriptor"));

var _decorators = require("../decorators");

var _collector = _interopRequireDefault(require("./collector"));

var _dec, _class;

var Logging = (_dec = (0, _decorators.deprecated)('Backendless.Logging', 'Backendless.Logging.flush'), (_class = /*#__PURE__*/function () {
  function Logging(app) {
    (0, _classCallCheck2["default"])(this, Logging);
    this.app = app;
    this.loggingCollector = new _collector["default"](app);
  }

  (0, _createClass2["default"])(Logging, [{
    key: "getLogger",
    value: function getLogger() {
      var _this$loggingCollecto;

      return (_this$loggingCollecto = this.loggingCollector).getLogger.apply(_this$loggingCollecto, arguments);
    }
  }, {
    key: "flushSync",
    value: function flushSync() {
      var _this$loggingCollecto2;

      return (_this$loggingCollecto2 = this.loggingCollector).flushSync.apply(_this$loggingCollecto2, arguments);
    }
  }, {
    key: "flush",
    value: function flush() {
      var _this$loggingCollecto3;

      return (_this$loggingCollecto3 = this.loggingCollector).flush.apply(_this$loggingCollecto3, arguments);
    }
  }, {
    key: "reset",
    value: function reset() {
      var _this$loggingCollecto4;

      return (_this$loggingCollecto4 = this.loggingCollector).reset.apply(_this$loggingCollecto4, arguments);
    }
  }, {
    key: "setLogReportingPolicy",
    value: function setLogReportingPolicy() {
      var _this$loggingCollecto5;

      return (_this$loggingCollecto5 = this.loggingCollector).setLogReportingPolicy.apply(_this$loggingCollecto5, arguments);
    }
  }]);
  return Logging;
}(), ((0, _applyDecoratedDescriptor2["default"])(_class.prototype, "flushSync", [_dec], Object.getOwnPropertyDescriptor(_class.prototype, "flushSync"), _class.prototype)), _class));
var _default = Logging;
exports["default"] = _default;
