// backendless/es/bl/invoke-service-method.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.invokeServiceMethod = invokeServiceMethod;

var _async = _interopRequireDefault(require("../request/async"));

var _constants = require("./constants");

function invokeServiceMethod(serviceName, method, parameters, executionType, asyncHandler) {
  if (typeof parameters === 'string' && (0, _constants.isExecutionType)(parameters)) {
    asyncHandler = executionType;
    executionType = parameters;
    parameters = undefined;
  }

  if (executionType instanceof _async["default"]) {
    asyncHandler = executionType;
    executionType = undefined;
  }

  if (parameters instanceof _async["default"]) {
    asyncHandler = parameters;
    parameters = undefined;
  }

  var headers = {};

  if (executionType) {
    headers[_constants.EXECUTION_TYPE_HEADER] = executionType;
  }

  return this.app.request.post({
    url: this.app.urls.blServiceMethod(serviceName, method),
    data: parameters,
    isAsync: !!asyncHandler,
    headers: headers,
    asyncHandler: asyncHandler
  });
}
