// backendless/es/users/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _applyDecoratedDescriptor2 = _interopRequireDefault(require("@babel/runtime/helpers/applyDecoratedDescriptor"));

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _utils = _interopRequireDefault(require("../utils"));

var _decorators = require("../decorators");

var _register = require("./register");

var _roles = require("./roles");

var _login = require("./login");

var _loginAsGuest = require("./login-as-guest");

var _logout = require("./logout");

var _update = require("./update");

var _describeClass = require("./describe-class");

var _restorePassword = require("./restore-password");

var _emailConfirmation = require("./email-confirmation");

var _status = require("./status");

var _social = require("./social");

var _currentUser = require("./current-user");

var _dec, _dec2, _dec3, _dec4, _dec5, _dec6, _dec7, _dec8, _dec9, _dec10, _dec11, _dec12, _dec13, _dec14, _dec15, _dec16, _obj, _init, _init2, _init3, _init4, _init5, _init6, _init7, _init8, _init9, _init10, _init11, _init12, _init13, _init14, _init15, _init16;

var Users = /*#__PURE__*/function () {
  function Users(app) {
    (0, _classCallCheck2["default"])(this, Users);
    this.app = app;
  }

  (0, _createClass2["default"])(Users, [{
    key: "getCurrentUserId",
    value: function getCurrentUserId() {
      if (this.currentUser) {
        return this.currentUser.objectId;
      }

      return this.app.LocalCache.get(this.app.LocalCache.Keys.CURRENT_USER_ID) || null;
    }
  }]);
  return Users;
}();

(0, _extends2["default"])(Users.prototype, (_dec = (0, _decorators.deprecated)('Backendless.Users', 'Backendless.Users.register'), _dec2 = (0, _decorators.deprecated)('Backendless.Users', 'Backendless.Users.getUserRoles'), _dec3 = (0, _decorators.deprecated)('Backendless.Users', 'Backendless.Users.assignRole'), _dec4 = (0, _decorators.deprecated)('Backendless.Users', 'Backendless.Users.unassignRole'), _dec5 = (0, _decorators.deprecated)('Backendless.Users', 'Backendless.Users.login'), _dec6 = (0, _decorators.deprecated)('Backendless.Users', 'Backendless.Users.loginAsGuest'), _dec7 = (0, _decorators.deprecated)('Backendless.Users', 'Backendless.Users.describeUserClass'), _dec8 = (0, _decorators.deprecated)('Backendless.Users', 'Backendless.Users.restorePassword'), _dec9 = (0, _decorators.deprecated)('Backendless.Users', 'Backendless.Users.logout'), _dec10 = (0, _decorators.deprecated)('Backendless.Users', 'Backendless.Users.getCurrentUser'), _dec11 = (0, _decorators.deprecated)('Backendless.Users', 'Backendless.Users.update'), _dec12 = (0, _decorators.deprecated)('Backendless.Users', 'Backendless.Users.loginWithFacebook'), _dec13 = (0, _decorators.deprecated)('Backendless.Users', 'Backendless.Users.loginWithGooglePlus'), _dec14 = (0, _decorators.deprecated)('Backendless.Users', 'Backendless.Users.loginWithTwitter'), _dec15 = (0, _decorators.deprecated)('Backendless.Users', 'Backendless.Users.isValidLogin'), _dec16 = (0, _decorators.deprecated)('Backendless.Users', 'Backendless.Users.resendEmailConfirmation'), (_obj = {
  registerSync: _utils["default"]["synchronized"](_register.register),
  register: _utils["default"].promisified(_register.register),
  getUserRolesSync: _utils["default"]["synchronized"](_roles.getUserRoles),
  getUserRoles: _utils["default"].promisified(_roles.getUserRoles),
  assignRoleSync: _utils["default"]["synchronized"](_roles.assignRole),
  assignRole: _utils["default"].promisified(_roles.assignRole),
  unassignRoleSync: _utils["default"]["synchronized"](_roles.unassignRole),
  unassignRole: _utils["default"].promisified(_roles.unassignRole),
  loginSync: _utils["default"]["synchronized"](_login.login),
  login: _utils["default"].promisified(_login.login),
  loginAsGuestSync: _utils["default"]["synchronized"](_loginAsGuest.loginAsGuest),
  loginAsGuest: _utils["default"].promisified(_loginAsGuest.loginAsGuest),
  describeUserClassSync: _utils["default"]["synchronized"](_describeClass.describeUserClass),
  describeUserClass: _utils["default"].promisified(_describeClass.describeUserClass),
  restorePasswordSync: _utils["default"]["synchronized"](_restorePassword.restorePassword),
  restorePassword: _utils["default"].promisified(_restorePassword.restorePassword),
  logoutSync: _utils["default"]["synchronized"](_logout.logout),
  logout: _utils["default"].promisified(_logout.logout),
  getCurrentUserSync: _utils["default"]["synchronized"](_currentUser.getCurrentUser),
  getCurrentUser: _utils["default"].promisified(_currentUser.getCurrentUser),
  updateSync: _utils["default"]["synchronized"](_update.update),
  update: _utils["default"].promisified(_update.update),
  loginWithFacebookSync: _utils["default"]["synchronized"](_social.loginWithFacebook),
  loginWithFacebook: _utils["default"].promisified(_social.loginWithFacebook),
  loginWithFacebookSdk: _social.loginWithFacebookSdk,
  loginWithGooglePlusSync: _utils["default"]["synchronized"](_social.loginWithGooglePlus),
  loginWithGooglePlus: _utils["default"].promisified(_social.loginWithGooglePlus),
  loginWithGooglePlusSdk: _social.loginWithGooglePlusSdk,
  loginWithTwitterSync: _utils["default"]["synchronized"](_social.loginWithTwitter),
  loginWithTwitter: _utils["default"].promisified(_social.loginWithTwitter),
  isValidLoginSync: _utils["default"]["synchronized"](_currentUser.isValidLogin),
  isValidLogin: _utils["default"].promisified(_currentUser.isValidLogin),
  resendEmailConfirmationSync: _utils["default"]["synchronized"](_emailConfirmation.resendEmailConfirmation),
  resendEmailConfirmation: _utils["default"].promisified(_emailConfirmation.resendEmailConfirmation),
  loggedInUser: _currentUser.loggedInUser,
  getCurrentUserToken: _currentUser.getCurrentUserToken,
  setCurrentUserToken: _currentUser.setCurrentUserToken,
  getLocalCurrentUser: _currentUser.getLocalCurrentUser,
  setLocalCurrentUser: _currentUser.setLocalCurrentUser,
  enableUser: _utils["default"].promisified(_status.enableUser),
  disableUser: _utils["default"].promisified(_status.disableUser)
}, ((0, _applyDecoratedDescriptor2["default"])(_obj, "registerSync", [_dec], (_init = Object.getOwnPropertyDescriptor(_obj, "registerSync"), _init = _init ? _init.value : undefined, {
  enumerable: true,
  configurable: true,
  writable: true,
  initializer: function initializer() {
    return _init;
  }
}), _obj), (0, _applyDecoratedDescriptor2["default"])(_obj, "getUserRolesSync", [_dec2], (_init2 = Object.getOwnPropertyDescriptor(_obj, "getUserRolesSync"), _init2 = _init2 ? _init2.value : undefined, {
  enumerable: true,
  configurable: true,
  writable: true,
  initializer: function initializer() {
    return _init2;
  }
}), _obj), (0, _applyDecoratedDescriptor2["default"])(_obj, "assignRoleSync", [_dec3], (_init3 = Object.getOwnPropertyDescriptor(_obj, "assignRoleSync"), _init3 = _init3 ? _init3.value : undefined, {
  enumerable: true,
  configurable: true,
  writable: true,
  initializer: function initializer() {
    return _init3;
  }
}), _obj), (0, _applyDecoratedDescriptor2["default"])(_obj, "unassignRoleSync", [_dec4], (_init4 = Object.getOwnPropertyDescriptor(_obj, "unassignRoleSync"), _init4 = _init4 ? _init4.value : undefined, {
  enumerable: true,
  configurable: true,
  writable: true,
  initializer: function initializer() {
    return _init4;
  }
}), _obj), (0, _applyDecoratedDescriptor2["default"])(_obj, "loginSync", [_dec5], (_init5 = Object.getOwnPropertyDescriptor(_obj, "loginSync"), _init5 = _init5 ? _init5.value : undefined, {
  enumerable: true,
  configurable: true,
  writable: true,
  initializer: function initializer() {
    return _init5;
  }
}), _obj), (0, _applyDecoratedDescriptor2["default"])(_obj, "loginAsGuestSync", [_dec6], (_init6 = Object.getOwnPropertyDescriptor(_obj, "loginAsGuestSync"), _init6 = _init6 ? _init6.value : undefined, {
  enumerable: true,
  configurable: true,
  writable: true,
  initializer: function initializer() {
    return _init6;
  }
}), _obj), (0, _applyDecoratedDescriptor2["default"])(_obj, "describeUserClassSync", [_dec7], (_init7 = Object.getOwnPropertyDescriptor(_obj, "describeUserClassSync"), _init7 = _init7 ? _init7.value : undefined, {
  enumerable: true,
  configurable: true,
  writable: true,
  initializer: function initializer() {
    return _init7;
  }
}), _obj), (0, _applyDecoratedDescriptor2["default"])(_obj, "restorePasswordSync", [_dec8], (_init8 = Object.getOwnPropertyDescriptor(_obj, "restorePasswordSync"), _init8 = _init8 ? _init8.value : undefined, {
  enumerable: true,
  configurable: true,
  writable: true,
  initializer: function initializer() {
    return _init8;
  }
}), _obj), (0, _applyDecoratedDescriptor2["default"])(_obj, "logoutSync", [_dec9], (_init9 = Object.getOwnPropertyDescriptor(_obj, "logoutSync"), _init9 = _init9 ? _init9.value : undefined, {
  enumerable: true,
  configurable: true,
  writable: true,
  initializer: function initializer() {
    return _init9;
  }
}), _obj), (0, _applyDecoratedDescriptor2["default"])(_obj, "getCurrentUserSync", [_dec10], (_init10 = Object.getOwnPropertyDescriptor(_obj, "getCurrentUserSync"), _init10 = _init10 ? _init10.value : undefined, {
  enumerable: true,
  configurable: true,
  writable: true,
  initializer: function initializer() {
    return _init10;
  }
}), _obj), (0, _applyDecoratedDescriptor2["default"])(_obj, "updateSync", [_dec11], (_init11 = Object.getOwnPropertyDescriptor(_obj, "updateSync"), _init11 = _init11 ? _init11.value : undefined, {
  enumerable: true,
  configurable: true,
  writable: true,
  initializer: function initializer() {
    return _init11;
  }
}), _obj), (0, _applyDecoratedDescriptor2["default"])(_obj, "loginWithFacebookSync", [_dec12], (_init12 = Object.getOwnPropertyDescriptor(_obj, "loginWithFacebookSync"), _init12 = _init12 ? _init12.value : undefined, {
  enumerable: true,
  configurable: true,
  writable: true,
  initializer: function initializer() {
    return _init12;
  }
}), _obj), (0, _applyDecoratedDescriptor2["default"])(_obj, "loginWithGooglePlusSync", [_dec13], (_init13 = Object.getOwnPropertyDescriptor(_obj, "loginWithGooglePlusSync"), _init13 = _init13 ? _init13.value : undefined, {
  enumerable: true,
  configurable: true,
  writable: true,
  initializer: function initializer() {
    return _init13;
  }
}), _obj), (0, _applyDecoratedDescriptor2["default"])(_obj, "loginWithTwitterSync", [_dec14], (_init14 = Object.getOwnPropertyDescriptor(_obj, "loginWithTwitterSync"), _init14 = _init14 ? _init14.value : undefined, {
  enumerable: true,
  configurable: true,
  writable: true,
  initializer: function initializer() {
    return _init14;
  }
}), _obj), (0, _applyDecoratedDescriptor2["default"])(_obj, "isValidLoginSync", [_dec15], (_init15 = Object.getOwnPropertyDescriptor(_obj, "isValidLoginSync"), _init15 = _init15 ? _init15.value : undefined, {
  enumerable: true,
  configurable: true,
  writable: true,
  initializer: function initializer() {
    return _init15;
  }
}), _obj), (0, _applyDecoratedDescriptor2["default"])(_obj, "resendEmailConfirmationSync", [_dec16], (_init16 = Object.getOwnPropertyDescriptor(_obj, "resendEmailConfirmationSync"), _init16 = _init16 ? _init16.value : undefined, {
  enumerable: true,
  configurable: true,
  writable: true,
  initializer: function initializer() {
    return _init16;
  }
}), _obj)), _obj)));
var _default = Users;
exports["default"] = _default;
