// backendless/es/files/persmissions/persmission.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _applyDecoratedDescriptor2 = _interopRequireDefault(require("@babel/runtime/helpers/applyDecoratedDescriptor"));

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _utils = _interopRequireDefault(require("../../utils"));

var _decorators = require("../../decorators");

var _sendRequest = require("./send-request");

var _dec, _dec2, _dec3, _dec4, _dec5, _dec6, _dec7, _dec8, _dec9, _dec10, _dec11, _dec12, _obj, _init, _init2, _init3, _init4, _init5, _init6, _init7, _init8, _init9, _init10, _init11, _init12;

var PermissionTypes = {
  GRANT: 'GRANT',
  DENY: 'DENY'
};

function grantForUser(userId, url, asyncHandler) {
  return _sendRequest.sendRequest.call(this, PermissionTypes.GRANT, url, {
    userId: userId
  }, asyncHandler);
}

function denyForUser(userId, url, asyncHandler) {
  return _sendRequest.sendRequest.call(this, PermissionTypes.DENY, url, {
    userId: userId
  }, asyncHandler);
}

function grantForRole(roleName, url, asyncHandler) {
  return _sendRequest.sendRequest.call(this, PermissionTypes.GRANT, url, {
    roleName: roleName
  }, asyncHandler);
}

function denyForRole(roleName, url, asyncHandler) {
  return _sendRequest.sendRequest.call(this, PermissionTypes.DENY, url, {
    roleName: roleName
  }, asyncHandler);
}

function grantForAllUsers(url, asyncHandler) {
  return _sendRequest.sendRequest.call(this, PermissionTypes.GRANT, url, {
    userId: '*'
  }, asyncHandler);
}

function denyForAllUsers(url, asyncHandler) {
  return _sendRequest.sendRequest.call(this, PermissionTypes.DENY, url, {
    userId: '*'
  }, asyncHandler);
}

function grantForAllRoles(url, asyncHandler) {
  return _sendRequest.sendRequest.call(this, PermissionTypes.GRANT, url, {
    roleName: '*'
  }, asyncHandler);
}

function denyForAllRoles(url, asyncHandler) {
  return _sendRequest.sendRequest.call(this, PermissionTypes.DENY, url, {
    roleName: '*'
  }, asyncHandler);
} //TODO: will be removed when remove sync methods


var namespaceLabel = 'Backendless.Files.Permissions.{READ|DELETE|WRITE}';

var FilePermission = function FilePermission(permission, app) {
  (0, _classCallCheck2["default"])(this, FilePermission);
  this.permission = permission;
  this.app = app;
};

exports["default"] = FilePermission;
(0, _extends2["default"])(FilePermission.prototype, (_dec = (0, _decorators.deprecated)(namespaceLabel, "".concat(namespaceLabel, ".grantForUser")), _dec2 = (0, _decorators.deprecated)(namespaceLabel, "".concat(namespaceLabel, ".grantForUser")), _dec3 = (0, _decorators.deprecated)(namespaceLabel, "".concat(namespaceLabel, ".denyForUser")), _dec4 = (0, _decorators.deprecated)(namespaceLabel, "".concat(namespaceLabel, ".denyForUser")), _dec5 = (0, _decorators.deprecated)(namespaceLabel, "".concat(namespaceLabel, ".grantForRole")), _dec6 = (0, _decorators.deprecated)(namespaceLabel, "".concat(namespaceLabel, ".grantForRole")), _dec7 = (0, _decorators.deprecated)(namespaceLabel, "".concat(namespaceLabel, ".denyForRole")), _dec8 = (0, _decorators.deprecated)(namespaceLabel, "".concat(namespaceLabel, ".denyForRole")), _dec9 = (0, _decorators.deprecated)(namespaceLabel, "".concat(namespaceLabel, ".grantForAllUsers")), _dec10 = (0, _decorators.deprecated)(namespaceLabel, "".concat(namespaceLabel, ".grantForAllUsers")), _dec11 = (0, _decorators.deprecated)(namespaceLabel, "".concat(namespaceLabel, ".denyForAllUsers")), _dec12 = (0, _decorators.deprecated)(namespaceLabel, "".concat(namespaceLabel, ".denyForAllUsers")), (_obj = {
  grantUserSync: _utils["default"]["synchronized"](grantForUser),
  grantUser: _utils["default"].promisified(grantForUser),
  denyUserSync: _utils["default"]["synchronized"](denyForUser),
  denyUser: _utils["default"].promisified(denyForUser),
  grantRoleSync: _utils["default"]["synchronized"](grantForRole),
  grantRole: _utils["default"].promisified(grantForRole),
  denyRoleSync: _utils["default"]["synchronized"](denyForRole),
  denyRole: _utils["default"].promisified(denyForRole),
  grantSync: _utils["default"]["synchronized"](grantForAllUsers),
  grant: _utils["default"].promisified(grantForAllUsers),
  denySync: _utils["default"]["synchronized"](denyForAllUsers),
  deny: _utils["default"].promisified(denyForAllUsers),
  grantForUser: _utils["default"].promisified(grantForUser),
  denyForUser: _utils["default"].promisified(denyForUser),
  grantForRole: _utils["default"].promisified(grantForRole),
  denyForRole: _utils["default"].promisified(denyForRole),
  grantForAllUsers: _utils["default"].promisified(grantForAllUsers),
  denyForAllUsers: _utils["default"].promisified(denyForAllUsers),
  grantForAllRoles: _utils["default"].promisified(grantForAllRoles),
  denyForAllRoles: _utils["default"].promisified(denyForAllRoles)
}, ((0, _applyDecoratedDescriptor2["default"])(_obj, "grantUserSync", [_dec], (_init = Object.getOwnPropertyDescriptor(_obj, "grantUserSync"), _init = _init ? _init.value : undefined, {
  enumerable: true,
  configurable: true,
  writable: true,
  initializer: function initializer() {
    return _init;
  }
}), _obj), (0, _applyDecoratedDescriptor2["default"])(_obj, "grantUser", [_dec2], (_init2 = Object.getOwnPropertyDescriptor(_obj, "grantUser"), _init2 = _init2 ? _init2.value : undefined, {
  enumerable: true,
  configurable: true,
  writable: true,
  initializer: function initializer() {
    return _init2;
  }
}), _obj), (0, _applyDecoratedDescriptor2["default"])(_obj, "denyUserSync", [_dec3], (_init3 = Object.getOwnPropertyDescriptor(_obj, "denyUserSync"), _init3 = _init3 ? _init3.value : undefined, {
  enumerable: true,
  configurable: true,
  writable: true,
  initializer: function initializer() {
    return _init3;
  }
}), _obj), (0, _applyDecoratedDescriptor2["default"])(_obj, "denyUser", [_dec4], (_init4 = Object.getOwnPropertyDescriptor(_obj, "denyUser"), _init4 = _init4 ? _init4.value : undefined, {
  enumerable: true,
  configurable: true,
  writable: true,
  initializer: function initializer() {
    return _init4;
  }
}), _obj), (0, _applyDecoratedDescriptor2["default"])(_obj, "grantRoleSync", [_dec5], (_init5 = Object.getOwnPropertyDescriptor(_obj, "grantRoleSync"), _init5 = _init5 ? _init5.value : undefined, {
  enumerable: true,
  configurable: true,
  writable: true,
  initializer: function initializer() {
    return _init5;
  }
}), _obj), (0, _applyDecoratedDescriptor2["default"])(_obj, "grantRole", [_dec6], (_init6 = Object.getOwnPropertyDescriptor(_obj, "grantRole"), _init6 = _init6 ? _init6.value : undefined, {
  enumerable: true,
  configurable: true,
  writable: true,
  initializer: function initializer() {
    return _init6;
  }
}), _obj), (0, _applyDecoratedDescriptor2["default"])(_obj, "denyRoleSync", [_dec7], (_init7 = Object.getOwnPropertyDescriptor(_obj, "denyRoleSync"), _init7 = _init7 ? _init7.value : undefined, {
  enumerable: true,
  configurable: true,
  writable: true,
  initializer: function initializer() {
    return _init7;
  }
}), _obj), (0, _applyDecoratedDescriptor2["default"])(_obj, "denyRole", [_dec8], (_init8 = Object.getOwnPropertyDescriptor(_obj, "denyRole"), _init8 = _init8 ? _init8.value : undefined, {
  enumerable: true,
  configurable: true,
  writable: true,
  initializer: function initializer() {
    return _init8;
  }
}), _obj), (0, _applyDecoratedDescriptor2["default"])(_obj, "grantSync", [_dec9], (_init9 = Object.getOwnPropertyDescriptor(_obj, "grantSync"), _init9 = _init9 ? _init9.value : undefined, {
  enumerable: true,
  configurable: true,
  writable: true,
  initializer: function initializer() {
    return _init9;
  }
}), _obj), (0, _applyDecoratedDescriptor2["default"])(_obj, "grant", [_dec10], (_init10 = Object.getOwnPropertyDescriptor(_obj, "grant"), _init10 = _init10 ? _init10.value : undefined, {
  enumerable: true,
  configurable: true,
  writable: true,
  initializer: function initializer() {
    return _init10;
  }
}), _obj), (0, _applyDecoratedDescriptor2["default"])(_obj, "denySync", [_dec11], (_init11 = Object.getOwnPropertyDescriptor(_obj, "denySync"), _init11 = _init11 ? _init11.value : undefined, {
  enumerable: true,
  configurable: true,
  writable: true,
  initializer: function initializer() {
    return _init11;
  }
}), _obj), (0, _applyDecoratedDescriptor2["default"])(_obj, "deny", [_dec12], (_init12 = Object.getOwnPropertyDescriptor(_obj, "deny"), _init12 = _init12 ? _init12.value : undefined, {
  enumerable: true,
  configurable: true,
  writable: true,
  initializer: function initializer() {
    return _init12;
  }
}), _obj)), _obj)));
