// backendless/es/cache/put.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.put = put;

var _utils = _interopRequireDefault(require("../utils"));

var _async = _interopRequireDefault(require("../request/async"));

function put(key, value, timeToLive, asyncHandler) {
  if (!key || !_utils["default"].isString(key)) {
    throw new Error('Cache Key must be non empty String');
  }

  if (timeToLive instanceof _async["default"]) {
    asyncHandler = timeToLive;
    timeToLive = undefined;
  }

  if (timeToLive && !_utils["default"].isNumber(timeToLive)) {
    throw new Error('Cache timeToLive must be Number');
  }

  if (_utils["default"].isObject(value) && value.constructor !== Object) {
    value.___class = value.___class || _utils["default"].getClassName(value);
  }

  if (asyncHandler) {
    asyncHandler = _utils["default"].wrapAsync(asyncHandler);
  }

  return this.app.request.put({
    url: this.app.urls.cacheItem(key) + (timeToLive ? '?timeout=' + timeToLive : ''),
    headers: {
      'Content-Type': 'application/json'
    },
    data: JSON.stringify(value),
    isAsync: !!asyncHandler,
    asyncHandler: asyncHandler
  });
}
