// backendless/lib/users/social/request.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.sendSocialLoginRequest = sendSocialLoginRequest;

var _async = _interopRequireDefault(require("../../request/async"));

var _utils = require("../utils");

function sendSocialLoginRequest(accessToken, socialType, fieldsMapping, stayLoggedIn, asyncHandler) {
  if (!accessToken) {
    return asyncHandler.fault('"accessToken" is missing.');
  }

  var context = this;
  var interimCallback = new _async["default"](function (r) {
    context.setLocalCurrentUser(_utils.parseResponse.call(context, r));
    context.app.LocalCache.set('stayLoggedIn', !!stayLoggedIn);
    asyncHandler.success(_utils.getUserFromResponse.call(context, context.getLocalCurrentUser()));
  }, function (e) {
    asyncHandler.fault(e);
  });
  this.app.request.post({
    url: this.app.urls.userSocialLogin(socialType),
    isAsync: true,
    asyncHandler: interimCallback,
    data: {
      accessToken: accessToken,
      fieldsMapping: fieldsMapping
    }
  });
}
