// backendless/lib/messaging/methods/send-email.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.sendEmail = sendEmail;

var _utils = _interopRequireDefault(require("../../utils"));

var _bodyParts = _interopRequireDefault(require("../helpers/body-parts"));

function sendEmail(subject, bodyParts, recipients, attachments
/**, async */
) {
  var responder = _utils["default"].extractResponder(arguments);

  var isAsync = !!responder;
  var data = {};

  if (subject && !_utils["default"].isEmpty(subject) && _utils["default"].isString(subject)) {
    data.subject = subject;
  } else {
    throw new Error('Subject is required parameter and must be a nonempty string');
  }

  if (bodyParts instanceof _bodyParts["default"] && !_utils["default"].isEmpty(bodyParts)) {
    data.bodyparts = bodyParts;
  } else {
    throw new Error('Use Bodyparts as bodyParts argument, must contain at least one property');
  }

  if (recipients && _utils["default"].isArray(recipients) && !_utils["default"].isEmpty(recipients)) {
    data.to = recipients;
  } else {
    throw new Error('Recipients is required parameter, must be a nonempty array');
  }

  if (attachments) {
    if (_utils["default"].isArray(attachments)) {
      if (!_utils["default"].isEmpty(attachments)) {
        data.attachment = attachments;
      }
    }
  }

  function responseMessageStatus(res) {
    return res.status;
  }

  return this.app.request.post({
    url: this.app.urls.messagingEmail(),
    isAsync: isAsync,
    asyncHandler: _utils["default"].wrapAsync(responder, responseMessageStatus),
    data: data
  });
}
