// backendless/lib/bl/dispatch-event.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.dispatchEvent = dispatchEvent;

var _utils = _interopRequireDefault(require("../utils"));

var _async = _interopRequireDefault(require("../request/async"));

var _constants = require("./constants");

function dispatchEvent(eventName, eventArgs, executionType, asyncHandler) {
  if (!eventName || !_utils["default"].isString(eventName)) {
    throw new Error('Event Name must be provided and must be not an empty STRING!');
  }

  if (typeof eventArgs === 'string' && (0, _constants.isExecutionType)(eventArgs)) {
    asyncHandler = executionType;
    executionType = eventArgs;
    eventArgs = undefined;
  }

  if (executionType instanceof _async["default"]) {
    asyncHandler = executionType;
    executionType = undefined;
  }

  if (eventArgs instanceof _async["default"]) {
    asyncHandler = eventArgs;
    eventArgs = undefined;
  }

  if (!_utils["default"].isObject(eventArgs)) {
    eventArgs = {};
  }

  if (asyncHandler) {
    asyncHandler = _utils["default"].wrapAsync(asyncHandler);
  }

  var headers = {};

  if (executionType) {
    headers[_constants.EXECUTION_TYPE_HEADER] = executionType;
  }

  return this.app.request.post({
    url: this.app.urls.blEvent(eventName),
    data: eventArgs,
    isAsync: !!asyncHandler,
    headers: headers,
    asyncHandler: asyncHandler
  });
}
