// backendless/es/request/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _request = require("./request");

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { (0, _defineProperty2["default"])(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

var Methods = {
  GET: 'GET',
  POST: 'POST',
  PUT: 'PUT',
  DELETE: 'DELETE',
  PATCH: 'PATCH'
};

var Request = /*#__PURE__*/function () {
  function Request(app) {
    (0, _classCallCheck2["default"])(this, Request);
    this.app = app;
    this.Methods = Methods;
  }

  (0, _createClass2["default"])(Request, [{
    key: "send",
    value: function send() {
      for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
        args[_key] = arguments[_key];
      }

      return _request.sendRequest.call.apply(_request.sendRequest, [this].concat(args));
    }
  }, {
    key: "get",
    value: function get(config) {
      return this.send(_objectSpread(_objectSpread({}, config), {}, {
        method: Methods.GET
      }));
    }
  }, {
    key: "post",
    value: function post(config) {
      return this.send(_objectSpread(_objectSpread({}, config), {}, {
        method: Methods.POST
      }));
    }
  }, {
    key: "put",
    value: function put(config) {
      return this.send(_objectSpread(_objectSpread({}, config), {}, {
        method: Methods.PUT
      }));
    }
  }, {
    key: "delete",
    value: function _delete(config) {
      return this.send(_objectSpread(_objectSpread({}, config), {}, {
        method: Methods.DELETE
      }));
    }
  }]);
  return Request;
}();

exports["default"] = Request;
