// backendless/lib/users/update.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.update = update;

var _utils = _interopRequireDefault(require("../utils"));

var _async = _interopRequireDefault(require("../request/async"));

var _user = _interopRequireDefault(require("./user"));

function update(user
/** async */
) {
  var responder = _utils["default"].extractResponder(arguments);

  var isAsync = !!responder;
  var result = this.app.request.put({
    url: this.app.urls.userObject(user.objectId),
    isAsync: isAsync,
    asyncHandler: responder && wrapAsync(responder),
    data: user
  });
  return isAsync ? result : parseResponse(result);
}

function parseResponse(data) {
  return _utils["default"].deepExtend(new _user["default"](), data);
}

function wrapAsync(asyncHandler) {
  //TODO: will be removed after remove all the "{methodName}Sync" methods
  var onSuccess = function onSuccess(data) {
    return asyncHandler.success(parseResponse(data));
  };

  var onError = function onError(error) {
    return asyncHandler.fault(error);
  };

  return new _async["default"](onSuccess, onError);
}
